Helm chart storage_resize_mode mixed does not work

- **Which image of the operator are you using?** registry.opensource.zalan.do/acid/postgres-operator:v1.8.2
- **Where do you run it - cloud or metal? Kubernetes or OpenShift?** AWS K8s 
- **Are you running Postgres Operator in production?** yes
- **Type of issue?** question

Hi! I wanted to tune gp3 persistent volume iops and throughput.

Documentation at https://postgres-operator.readthedocs.io/en/latest/reference/operator_parameters/#kubernetes-resources
states that there is:
storage_resize_mode parameter - mixed :operator uses AWS API to adjust size, throughput, and IOPS, and calls pvc change for file system resize Default is "pvc".

When i add this parameter in postgres-operator 1.8.2 helm chart values i get error that such parameter is unsupported. 

Checked 1.8.2 helm chart crd manifest and such option "mixed" is missing - https://github.com/zalando/postgres-operator/blob/master/charts/postgres-operator/crds/operatorconfigurations.yaml#L320

Is iops/throughput tuning feature supported or not?

@FxKu Tried manual switch, but operator shows errors on sync volume:
time="2022-07-04T12:59:03Z" level=debug msg="syncing volumes using \"mixed\" storage resize mode" cluster-name=zabbix-stage/zabbix-psql-stage pkg=cluster
18
time="2022-07-04T12:59:03Z" level=info msg="starting reading ebs meta data" cluster-name=zabbix-stage/zabbix-psql-stage pkg=cluster
17
time="2022-07-04T12:59:03Z" level=debug msg="found 1 persistent volumes, size of known volumes 0" cluster-name=argo-watcher/argo-watcher-psql pkg=cluster
16
panic: runtime error: invalid memory address or nil pointer dereference
15
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x17df3fe]
14
13
goroutine 54 [running]:
12
github.com/zalando/postgres-operator/pkg/cluster.(*Cluster).populateVolumeMetaData(0xc000766000)
11
	/workspace/pkg/cluster/volumes.go:160 +0x25e
10
github.com/zalando/postgres-operator/pkg/cluster.(*Cluster).syncVolumes(0xc000766000)
9
	/workspace/pkg/cluster/volumes.go:36 +0x135
8
github.com/zalando/postgres-operator/pkg/cluster.(*Cluster).Sync(0xc000766000, 0xc0007e0480)
7
	/workspace/pkg/cluster/sync.go:68 +0x3f2
6
github.com/zalando/postgres-operator/pkg/controller.(*Controller).processEvent(0xc0001d2380, {{0xc0a8d6e11819f1b4, 0x1021712e3, 0x30b9b20}, {0xc000498c90, 0x24}, {0x1d107b8, 0x4}, 0x0, 0xc0007e0480, ...})
5
	/workspace/pkg/controller/postgresql.go:310 +0xe90
4
github.com/zalando/postgres-operator/pkg/controller.(*Controller).processClusterEventsQueue(0xc0001d2380, 0x5, 0xc000114f00, 0x0)
3
	/workspace/pkg/controller/postgresql.go:344 +0x2c5
2
created by github.com/zalando/postgres-operator/pkg/controller.(*Controller).Run
1
	/workspace/pkg/controller/controller.go:446 +0x51

--- a/charts/postgres-operator/crds/operatorconfigurations.yaml
+++ b/charts/postgres-operator/crds/operatorconfigurations.yaml
@@ -321,6 +321,7 @@ spec:
                     type: string
                     enum:
                       - "ebs"
+                      - "mixed"
                       - "pvc"
                       - "off"
                     default: "pvc"

--- a/charts/postgres-operator/values.yaml
+++ b/charts/postgres-operator/values.yaml
@@ -196,7 +196,7 @@ configKubernetes:
   # whether the Spilo container should run with additional permissions other than parent.
   # required by cron which needs setuid
   spilo_allow_privilege_escalation: true
-  # storage resize strategy, available options are: ebs, pvc, off
+  # storage resize strategy, available options are: ebs, pvc, off or mixed
   storage_resize_mode: pvc
   # pod toleration assigned to instances of every Postgres cluster
   # toleration:

--- a/docs/reference/operator_parameters.md
+++ b/docs/reference/operator_parameters.md
@@ -486,10 +486,10 @@ configuration they are grouped under the `kubernetes` key.
 * **storage_resize_mode**
   defines how operator handles the difference between the requested volume size and
     the actual size. Available options are:
-    1. `ebs` : operator resizes EBS volumes directly and executes `resizefs` within a pod
-    2. `pvc` : operator only changes PVC definition
-    3. `off` : disables resize of the volumes.
-    4. `mixed` :operator  uses AWS API to adjust size, throughput, and IOPS, and calls pvc change for file system resize
+    1. `ebs`   : operator resizes EBS volumes directly and executes `resizefs` within a pod
+    2. `pvc`   : operator only changes PVC definition
+    3. `off`   : disables resize of the volumes.
+    4. `mixed` : operator uses AWS API to adjust size, throughput, and IOPS, and calls pvc change for file system resize
     Default is "pvc".
 
 ## Kubernetes resource requests

--- a/docs/user.md
+++ b/docs/user.md
@@ -1029,9 +1029,9 @@ specified but globally disabled in the configuration. The
 
 ## Increase volume size
 
-Postgres operator supports statefulset volume resize if you're using the
-operator on top of AWS. For that you need to change the size field of the
-volume description in the cluster manifest and apply the change:
+Postgres operator supports statefulset volume resize without doing a rolling
+update. For that you need to change the size field of the volume description
+in the cluster manifest and apply the change:
 
 ```yaml
 spec:
@@ -1040,22 +1040,29 @@ spec:
 ```
 
 The operator compares the new value of the size field with the previous one and
-acts on differences.
+acts on differences. The `storage_resize_mode` can be configured. By default,
+the operator will adjust the PVCs and leave it to K8s and the infrastructure to
+apply the change.
 
-You can only enlarge the volume with the process described above, shrinking is
-not supported and will emit a warning. After this update all the new volumes in
-the statefulset are allocated according to the new size. To enlarge persistent
-volumes attached to the running pods, the operator performs the following
-actions:
+When using AWS with gp3 volumes you should set the mode to `mixed` because it
+will also adjust the IOPS and throughput that can be defined in the manifest.
+Check the [AWS docs](https://aws.amazon.com/ebs/general-purpose/) to learn
+about default and maximum values. Keep in mind that AWS rate-limits updating
+volume specs to no more than once every 6 hours. 
 
-* call AWS API to change the volume size
+```yaml
+spec:
+  volume:
+    size: 5Gi # new volume size
+    iops: 4000
+    throughput: 500
+```
 
-* connect to pod using `kubectl exec` and resize filesystem with `resize2fs`.
-
-Fist step has a limitation, AWS rate-limits this operation to no more than once
-every 6 hours. Note, that if the statefulset is scaled down before resizing the
-new size is only applied to the volumes attached to the running pods. The
-size of volumes that correspond to the previously running pods is not changed.
+The operator can only enlarge volumes. Shrinking is not supported and will emit
+a warning. However, it can be done manually after updating the manifest. You
+have to delete the PVC, which will hang until you also delete the corresponding
+pod. Proceed with the next pod when the cluster is healthy again and replicas
+are streaming.
 
 ## Logical backups
 

--- a/manifests/operatorconfiguration.crd.yaml
+++ b/manifests/operatorconfiguration.crd.yaml
@@ -319,6 +319,7 @@ spec:
                     type: string
                     enum:
                       - "ebs"
+                      - "mixed"
                       - "pvc"
                       - "off"
                     default: "pvc"

--- a/pkg/apis/acid.zalan.do/v1/crds.go
+++ b/pkg/apis/acid.zalan.do/v1/crds.go
@@ -1429,6 +1429,9 @@ var OperatorConfigCRDResourceValidation = apiextv1.CustomResourceValidation{
 										Raw: []byte(`"ebs"`),
 									},
 									{
+										Raw: []byte(`"mixed"`),
+									},
+									{
 										Raw: []byte(`"pvc"`),
 									},
 									{

